diff --git a/pandas/tests/test_downstream.py b/pandas/tests/test_downstream.py
index 9fe8b0f95..f1ee5723d 100644
--- a/pandas/tests/test_downstream.py
+++ b/pandas/tests/test_downstream.py
@@ -135,11 +135,15 @@ def test_pyarrow(df):
 
 def test_missing_required_dependency():
     # GH 23868
-    # use the -S flag to disable site-packages
-    call = ['python', '-S', '-c', 'import pandas']
+    # To ensure proper isolation, we pass these flags
+    # -S : disable site-packages
+    # -s : disable user site-packages
+    # -E : disable PYTHON* env vars, especially PYTHONPATH
+    call = ['python', '-S', '-s', '-E', '-c', 'import pandas']
 
     with pytest.raises(subprocess.CalledProcessError) as exc:
         subprocess.check_output(call, stderr=subprocess.STDOUT)
 
     output = exc.value.stdout.decode()
+    assert output == b'123'
     assert all(x in output for x in ['numpy', 'pytz', 'dateutil'])
