diff --git a/pandas/tests/test_downstream.py b/pandas/tests/test_downstream.py
index 9fe8b0f95..9be107f9c 100644
--- a/pandas/tests/test_downstream.py
+++ b/pandas/tests/test_downstream.py
@@ -135,8 +135,23 @@ def test_pyarrow(df):
 
 def test_missing_required_dependency():
     # GH 23868
-    # use the -S flag to disable site-packages
-    call = ['python', '-S', '-c', 'import pandas']
+    # To ensure proper isolation, we pass these flags
+    # -S : disable site-packages
+    # -s : disable user site-packages
+    # -E : disable PYTHON* env vars, especially PYTHONPATH
+    # And, that's apparently not enough, so we give up.
+    # https://github.com/MacPython/pandas-wheels/pull/50
+    try:
+        subprocess.check_output(['python', '-sSE', '-c', 'import numpy'],
+                                stderr=subprocess.DEVNULL)
+    except subprocess.CalledProcessError:
+        # NumPy is not around, we can do the test
+        pass
+    else:
+        # NumPy is in the isolation environment, give up.
+        pytest.skip("Required dependencies in isolated environment.")
+
+    call = ['python', '-sSE', '-c', 'import pandas']
 
     with pytest.raises(subprocess.CalledProcessError) as exc:
         subprocess.check_output(call, stderr=subprocess.STDOUT)
